pagefaultgames/pokerogue
# [Qol/Balance] Dynamax cannon tweak

## What are the changes?

- Changed Dynamax Cannon damage to scale off of how overleveled the target is in comparison to the maximum level you can be at that particular Wave. 

Maths: Now the full damage (200 BP) is reached when the target of the move is 5% (or more) __**above the level cap**__. 

If it's only 2.5% above the level cap, the damage is 150BP. At level cap or below, the target takes a 100BP move. With the newest changes, overleveling 20 or so levels won't automatically mean you'd be taking full damage (as it was the case with the previous 10 levels differential I proposed), but now you'd have to be the full 5% above it. ie. at level cap 3000, you'd need to be lvl 3150 to take full Dynamax Cannon damage.

I even made a nice spreadsheet for you to peruse the difference: https://docs.google.com/spreadsheets/d/1k6AuEBtP1J75iowQp5KNzAa05Lblq4_s32-Ti4B6OF8/edit?usp=sharing

- Changed the move's description to match (English locale)
- Added comments on the other locales so that translators know to check the new English description.

## Why am I doing these changes?

I discussed with balance team how it felt highly counter-intuitive that a +1 level bump near the final fight that would take a mon, or maybe even your entire team with a rarer candy, from 200 to 201 would mean that you take x2 when Eternatus decides to Dynamax Cannon. With this newer implementation, you'd have to bring a 210 mon to take the same amount of damage, which is a situation a new player is much less likely to encounter.

The arbitrary number of 200 also seemed very off with the rest of the design philosophy of this game.

Boss Eternatus should still be hitting as hard when it comes to players going hard on candy jars in Endless. Eternatus as a starter (and whatever he would be fused with) takes a bit of a hit as it won't have access to a 200BP on mons above lvl 200. This seems to me in line with normal move balancing, a 200BP move with no downsides is rare.



It's mainly oriented to give clarity and transparency, especially thinking of newer players who might try to do a setup strategy on Eternatus and not know of the 201+ level breakpoint -- which was a mechanic not from mainline games either.


### Screenshots/Videos

The screenshots here provide examples of how the damage multiplier works now.

Being at the level's cap will simply get you hit by 100BP
![Lvl 200 at Wave 200](https://github.com/pagefaultgames/pokerogue/assets/170796740/4809c391-0c10-4fa5-b01c-be732263a76e)

Being lvl 201 doesn't get you hit by 200BP, but instead it's a 10% increase when calculated at floor 200
![Lvl 201 at Wave 200](https://github.com/pagefaultgames/pokerogue/assets/170796740/ee1c7633-d89c-4ce1-925a-92cb179494d0)

Being at lvl 210 on wave 200 (5% over level cap) gets you hit for the full 200BP
![Lvl 210 at Wave 200](https://github.com/pagefaultgames/pokerogue/assets/170796740/40362a05-f472-4787-8150-151bc7059b7b)

The changes also work for lower level, were you ever face a low level Eternatus (?) or use Dynamax Cannon on overleveled foes below floor 200 (here max lvl is 56. so the extra overlevel from Blissey means about a 30% boost in the BP, even if its just one level.

![Lvl 57 at Wave 70](https://github.com/pagefaultgames/pokerogue/assets/170796740/80d0975f-7453-4680-82d8-db02a5393451)

Here Blissey is overleveled beyond belief for being floor 1, but damage caps at 200BP so that's what she's taking.
![Lvl 200 at Wave 1](https://github.com/pagefaultgames/pokerogue/assets/170796740/c06f10c8-b434-41b5-971c-d09effd13c25)


## How to test the changes?

Make an override, give your PKMN Dynamax Cannon and play around with level overrides. Make sure to use a SpDef Wall


## Checklist
